### PR TITLE
Neater UI

### LINKED ISSUE #170: Process Manager Layout and Design
Now that the main functionality is in place we should look at reviewing the layout of the page. This should include:
- maximising the space used on the screen - currently there is space at either side that is unused.
- reviewing the general elements used to in the layout and identifying if we can do anything in a more bootstrappy way.
- better managing the use of space between the process table and message feed e.g. capping the max width of the message feed. At the moment as content updates in the message feed and/or process table the column updates will change which looks a little janky.
- Look at setting vertical height of message feed more intelligently.

### PR DESCRIPTION
**Pull request by cvzbynek**
# Description

This PR improves the user interface for the process manager page, introducing several enhancements for a more polished, functional layout:
1. Adjusted the **message panel** to maximize screen space usage.
2. Implemented **Flexbox layout** for the main container to allow the process control table to expand to full width when needed.
3. Minor CSS adjustments to improve table and button spacing, uniform button appearance, and column alignment.
4. **Logs** screen revised

Fixes #170 

## Type of change

- [ ] Documentation (non-breaking change that adds or improves the documentation)
- [x] New feature (non-breaking change which adds functionality)
- [x] Optimization (non-breaking, back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [ ] All tests pass (eg. `python -m pytest`)
- [ ] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [ ] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added or an issue has been opened to tackle that in the future.

### CODE REVIEW

File: process_manager/tables.py [hunk 1]
```diff
@@ -1,61 +1,130 @@
...
 
-header_checkbox_hyperscript = "on click set .row-checkbox.checked to my.checked"
-
-row_checkbox_hyperscript = """
-on click
-if <.row-checkbox:not(:checked)/> is empty
-  set #header-checkbox.checked to true
-else
-  set #header-checkbox.checked to false
-"""
```
Comment (Sahil590) [inline process_manager/tables.py:18]: Why was this removed? 
The header checkbox does not work anymore.

**Review by Sahil590 — COMMENTED** [review]
Overall UI looks much better. 
I just had some comments on some hyperscript being omitted and some of the changes I made in PR #179 will conflict with yours but can address them once merged.

File: process_manager/tables.py (hunk 1 above)
Comment (cvzbynek) [reply process_manager/tables.py:18]: Sorry, I've lost it one the way. I will add it back

**Comment by jamesturner246** [comment]
I'll look into this soon, but for now I note that the messages feed is no longer populated with messages.

File: process_manager/templates/process_manager/index.html [hunk 2]
```diff
@@ -28,76 +66,87 @@
...
+              </div>
+              <input class="form-control mb-3"
+                     type="search"
+                     name="search"
+                     placeholder="Search processes..."
+                     hx-get="{% url 'process_manager:process_table' %}"
+                     hx-trigger="input changed delay:500ms, every 1s"
+                     hx-target="div.table-container">
+              <div id="table-container" class="table-container">
+                <div class="scrollable-div"></div>
```
Comment (cc-a) [inline process_manager/templates/process_manager/index.html:118]: This doesn't do anything as it will be replaced on initial load of the table.

File: process_manager/templates/process_manager/index.html [hunk 3]
```diff
@@ -28,76 +66,87 @@
...
+                        _="on click hide me toggle .hide-messages on #main-content">Show Messages</button>
+              </div>
+              <input class="form-control mb-3"
+                     type="search"
+                     name="search"
+                     placeholder="Search processes..."
+                     hx-get="{% url 'process_manager:process_table' %}"
+                     hx-trigger="input changed delay:500ms, every 1s"
+                     hx-target="div.table-container">
+              <div id="table-container" class="table-container">
```
Comment (cc-a) [inline process_manager/templates/process_manager/index.html:117]: The tables library already wraps the table in a separator div with `class="table-container"` so no need to apply here as well.

File: process_manager/views/pages.py [hunk 4]
```diff
@@ -11,12 +11,15 @@
...
 
 
 @login_required
 def index(request: HttpRequest) -> HttpResponse:
-    """View that renders the index/home page."""
-    return render(request=request, template_name="process_manager/index.html")
+    """View that renders the index/home page with process table."""
+    table_data: list[dict[str, str | int]] = []  # Placeholder for actual data retrieval
+    table = ProcessTable(table_data)  # Instantiate table with data
+    return render(request, "process_manager/index.html", {"table": table})
```
Comment (cc-a) [inline process_manager/views/pages.py:22]: I don't think this does anything as the table is not rendered in `index.html`.

File: process_manager/views/pages.py [hunk 5]
```diff
@@ -32,10 +35,12 @@ def logs(request: HttpRequest, uuid: uuid.UUID) -> HttpResponse:
...
     logs_response = get_process_logs(str(uuid))
-    context = dict(log_text="\n".join(val.data.line for val in logs_response))
-    return render(
-        request=request, context=context, template_name="process_manager/logs.html"
-    )
+
+    # Process the log text to exclude empty lines
+    log_lines = [val.data.line for val in logs_response if val.data.line.strip()]
+
+    context = {"log_lines": log_lines, "log_text": "\n".join(log_lines)}
```
Comment (cc-a) [inline process_manager/views/pages.py:42]: Seems `log_text` is no longer used in the template context and can be removed.

**Review by cc-a — CHANGES_REQUESTED** [review]
Thanks @cvzbynek. What a make over! Looks great. I've made a few comments where I things an be tidied up a bit. Could you also take a pass at the alignment of content in the table columns? Whilst the left align works well for the UUID column I think the others would benefit from having their headers and content centered.

File: process_manager/tables.py [hunk 6]
```diff
@@ -41,23 +63,39 @@ class Meta:  # noqa: D106
                 "id": "header-checkbox",
                 "hx-preserve": "true",
                 "_": header_checkbox_hyperscript,
-            }
+                "class": "form-check-input form-check-input-lg",
```
[the remark this answers is not in the document]
Comment (cvzbynek) [reply process_manager/tables.py:66]: Makes checkbox large. Comes from Bootstrap.

File: process_manager/templates/process_manager/index.html (hunk 2 above)
Comment (cvzbynek) [reply process_manager/templates/process_manager/index.html:118]: Do you mean scrollable-div? It is leftover from brief attempt to solve #182

File: process_manager/views/pages.py (hunk 5 above)
Comment (cvzbynek) [reply process_manager/views/pages.py]: `log_lines` gives more granular control for rendering each line, while `log_text` maintains compatibility with existing tests and code until it's fully transitioned to use `log_lines`.

File: process_manager/templates/process_manager/index.html (hunk 2 above)
Comment (cc-a) [reply process_manager/templates/process_manager/index.html]: Let's remove it then.

Comment (cvzbynek) [reply process_manager/templates/process_manager/index.html]: Already done :)

File: process_manager/views/pages.py (hunk 5 above)
Comment (cvzbynek) [reply process_manager/views/pages.py]: Ok, I'll remove it. Can you update the test?

Comment (cc-a) [reply process_manager/views/pages.py]: You can probably bundle it it's a simple change here to swap log_text for log_lines - https://github.com/ImperialCollegeLondon/drunc_ui/blob/99163d227022ad3d857ca15d87422795f8ca3412/tests/process_manager/views/test_page_views.py#L36

Comment (cvzbynek) [reply process_manager/views/pages.py]: Seems ok!

**Review by cc-a — APPROVED** [review]
Ok, ready to go I think.

**Review by jamesturner246 — APPROVED** [review]
LGTM. :+1: 

Not important for this PR, but maybe we can start to think about moving the CSS to dedicated documents.

File: process_manager/templates/process_manager/index.html [hunk 7]
```diff
@@ -1,18 +1,121 @@
...
       float: right;
     }
+    #show-messages-button {
+      display: none; /* Hide on load */
+    }
     #message-list {
       max-height: 80vh;
       overflow-y: auto;
     }
+    .table-container {
```
Comment (AdrianDAlessandro) [inline process_manager/templates/process_manager/index.html:19]: There is a lot of extra css in this PR, and it's all put into the index page for the process_manager. Would it be better off that it is moved to the `base.html` so all the pages (in both the process_manager and the controller) inherit the same css adjustments?
